src/main/java/com/codejoust/main/service/RoomService.java

```diff
@@ -363,6 +363,11 @@ public RoomDto setSpectator(String roomId, SetSpectatorRequest request) {
             throw new ApiException(RoomError.NOT_FOUND);
         }
 
+        // Return error if user changes status to non-spectator in active game.
+        if (room.getActive() && !request.getSpectator()) {
+            throw new ApiException(RoomError.ACTIVE_GAME);
+        }
+
```

Comment:
Are there any situations where we'd allow people currently competing to switch themselves into spectators?

Reply:
I couldn't really think of any, but thought the handicap would only occur when a spectator switches themselves to a non-spectator as they may have seen other people's code, which is why I prevented that here.

I could prevent the other switch too (non-spectator to spectator), but I figured I wouldn't overcomplicate the method without having a clear reason to prevent this functionality.